pkg/cloudprovider/providers/cloudstack/cloudstack.go

```diff
@@ -130,3 +133,17 @@ func (cs *CSCloud) GetZone() (cloudprovider.Zone, error) {
 	glog.V(2).Infof("Current zone is %v", cs.zone)
 	return cloudprovider.Zone{Region: cs.zone}, nil
 }
+
+// GetZoneByProviderID implements Zones.GetZoneByProviderID
+// This is particularly useful in external cloud providers where we can't rely on
```

Comment:
I'd change this to say `where the kubelet does not initialize node data`.`rely` to me makes it seem like it's possible for the kubelet to do it for external providers, but it's not.